### PR TITLE
refactor graphs implementation and add unit tests

### PR DESCRIPTION
**Pull request by rabi-siddique**
The PR does the following:
- Error handling mainly in `VaultCharts.ts` component. 
- Creates a utility function `extractDailyOracles` that is used in `VaultCharts.ts` and `ReserveHistoryGraph`
- Refactors graph data-related code into their separate functions. 
- Add unit tests to ensure the logic is null-safe and calculations are right. And avoid manual testing every time there is a code change in logic. 
- Updating types and chores where those types are being used.

### CODE REVIEW

File: tests/Vaults/daily-oracles.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      ATOM_oracle: {
+        nodes: [],
+      },
+    };
+
+    const result = extractDailyOracles(tokenName, dailyMetricsResponse);
+
+    expect(result).toEqual({});
+  });
+
```
Comment (frazarshad) [inline tests/Vaults/daily-oracles.spec.ts:28]: a more common scenarion would be :
```
  const dailyMetricsResponse = {
      ATOM_oracle: undefined,
    };
```
lets test for that as well

**Comment by frazarshad** [comment]
please re-verify your tests. they are failing

**Comment by rabi-siddique** [comment]
> please re-verify your tests. they are failing

Yeah, there was a recent update in `VaultCharts.tsx` for `totalDebtLast`:
```
    const totalDebtLast = dailyTokenMetrics?.totalDebtLast / 1_000_000;

``

Now, we divide it's value by `1_000_000`. 
I've updated the tests. Thanks for pointing it out

File: tests/Vaults/daily-oracles.spec.ts (hunk 1 above)
Comment (rabi-siddique) [reply tests/Vaults/daily-oracles.spec.ts:28]: Adding in #56

**Review by frazarshad — APPROVED** [review]
